relentless/optimize/__init__.py

```diff
@@ -1,3 +1,2 @@
-from .core import *
-from .algorithm import *
 from .objective import *
+from .optimizer import *
```

Comment:
When I refactored code, I have moved away from using `import *`, as it is better to explicit than implicit. A few things that we want to tweak related to this:
1. Our organization has been to implement a file `optimize.py` (`from . import optimize`) that contains the abstract base classes, then other modules with specific instances. You can explicitly import these into `__init__.py` using a tuple (go check one of the other modules).
2. Let's rename `optimizer` to `method`, I think this is cleaner.
3. Remove `from . import *` and replace by `from . import module_name`.
4. Remove `__all__` from the modules, as we are not doing any `import *` anywhere, so we don't really need these.